bftengine/src/bcstatetransfer/RVBManager.cpp

```diff
@@ -95,15 +107,21 @@ void RVBManager::init(bool fetching) {
     }
   }
 
-  if (!loaded_from_data_store && (desc.maxBlockId > 0)) {
+  // Do not attempt to build RVT in case
+  // 1. RVT is already build from RVB data stored in last checkpoint
+  // 2. Max Block Id from last stored checkpoint is 0
+  // 3. last reachable block == default genesis block (eg. 1)
+
+  if ((!loaded_from_data_store) && ((desc.maxBlockId) || (last_reachable_block_num > 1))) {
```

Comment:
1) Why do you put brackets around everything, e.g !loaded_from_data_store ? also you converted (desc.maxBlockId > 0) to (desc.maxBlockId ) which is more of a C style writing, and reduce readibility. It is less clearer in my opinion (you must relay on naming to understand the type, if we write all code like that it might force writer to go and check type, and even cause bugs).
What if i decide to change the type of x to int64_t and we have 'if (x)' on 1000 places (leave a side automatic replace). which ways of coding is better?

2) The above is not accurate.
You relay on the fact that we always have a non-pruned blockchain.
What if the same issue happens in a pruned blockchain?
You should get the genesis block id outside the condition:
```
auto genesis_block_id = as_->getGenesisBlockNum();
if (!loaded_from_data_store && ((desc.maxBlockId > 0) || (last_reachable_block_num > genesis_block_id))) {
...
```

Reply:
Thanks. Addressed.